### ISSUE TITLE
[TF 2.0 alpha] Erratic metric behaviour with MAE and MSE

### ISSUE THREAD

**Issue by roya0045** [tensorflow/tensorflow#29555]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below):2.0.0a0
- Python version:3.6.6
- Bazel version (if compiling from source):-
- GCC/Compiler version (if compiling from source):-
- CUDA/cuDNN version: Using CPU
- GPU model and memory: Using CPU

**Describe the current behavior**
The MAE and the MSE should be correlated but the MSE value used in the metric is not correlated [see this minimal log for instance](https://gist.github.com/roya0045/5e4c2950da48092af266c945dc69e0a1)

**Describe the expected behavior**
I assume that the MSE should be equivalent to the MAE squared but I may be wrong.

**Code to reproduce the issue**
Build a small LSTM with keras for sequence prediction.

Loss: 'mae'
metrics: ['mse','acc']
optimizer: kr.optimizers.Adam

**Other info / logs**
See link in the described behaviour.

**Comment by lufol** [comment]
Hi, looking at the source code you can see that they are correlated in some way. However, due to their differences, you can find data, such that they do not behave the same way.
Maybe you can have a look at [this post](https://medium.com/human-in-a-machine-world/mae-and-rmse-which-metric-is-better-e60ac3bde13d). They also show occasions, where both metrics show non correlating events.

**Comment by roya0045** [comment]
@lufol If you look more closely at the end of the 2nd epoch and the start of the 3rd epoch ( line 650 up to around line 828) you can see that the behaviour changes between epochs, which is highly unusual.

I forgot to add that my data are rescaled between 0 and 1. I also something get random spikes in MSA error (in the thousands) in half an epoch and the next pass everything goes back to normal.

**Comment by roya0045** [comment]
@lufol But you may be right and it might just be me overreacting. I'll close this in the meantime, thanks!

**Comment by lufol** [comment]
Yes @roya0045 , I get definitely your point. If you would be able to provide a small demo of the behaivour it would be much easier to serach for the cause.

**Comment by roya0045** [comment]
@lufol What would that small demo entail?

**Comment by lufol** [comment]
I guess by opening an issue here, you suspected that there is a bug somewhere in the code. By providing an easy to follow example that reproduces this bug, it would be much easier to find the cause for that.

Best regards

**Comment by roya0045** [comment]
@lufol My setup is quite simple, I have time series data in input, I have 4 lstm layers and the last one does not output a sequence but all others do. The output are the next 5 dates for x elements. Those 5 days are squashed. So if I want the output of 3 elements, I will have  15 output.

The compilation is as mentioned in the main post.

The layers are aroud 270 units wide.

The layers are not stateful (could this cause the behaviour change between epochs?)
